src/libYARP_dev/include/yarp/dev/IRobotDescription.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2016 iCub Facility, Istituto Italiano di Tecnologia
+ * Authors: Marco Randazzo <marco.randazzo@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#ifndef YARP_DEV_IROBOTDESCRIPTION_H
+#define YARP_DEV_IROBOTDESCRIPTION_H
+
+#include <yarp/os/Vocab.h>
+#include <yarp/os/Log.h>
+#include <vector>
+#include <string>
+
+namespace yarp {
+    namespace dev {
+        class IRobotDescription;
+
+        struct RobotDescription
+        {
+            std::string  device_name;
+            std::string  device_type;
+            bool RobotDescription::operator ==(RobotDescription const& b) { return this->device_name == b.device_name && this->device_type == b.device_type; }
+        };
+      }
+}
+
+/**
+ * @ingroup dev_iface_motor
+ *
+ * Control board, encoder interface.
+ */
+class yarp::dev::IRobotDescription
+{
+public:
+    /**
+     * Destructor.
+     */
+    virtual ~IRobotDescription() {}
+
+    /**
+     * Ask the complete list of all yarp device drivers registered by a robot description server.
+     */
+    virtual bool getAllDevices(std::vector<RobotDescription>& dev_list) = 0;
+
+    /**
+    * Ask a list of all yarp control board wrappers registered by a robot description server.
```

Comment:
It would be helpful to document here (I checked in the code) that "yarp control board wrappers" are the devices of type `controlboardwrapper` and `controlboardwrapper2`.